client/similargames.jsx

```diff
@@ -19,29 +19,75 @@ class SimilarGames extends React.Component  {
     super(props)
 
     this.state = {
+      gameID: 1,
       currentGame: [
-        {id: 1,
-        name: "Voom Voom Motorcycle",
-        price: 95,
-        sale_percent: 0.25,
-        series: "Product Creative Assistant",
-        seriesID: 1,
-        releasedt: "2020-02-14",
-        reviews: 143,
-        average_review: 4.5,
-        tags: "auxiliary, wireless , open-source, solid state"
+        {
+          id: 1,
+          name: "magnam quis et",
+          series: "quidem fuga blanditiis",
+          price: 60.41999816894531,
+          sale: 0.009999999776482582,
+          release: "2020-01-11",
+          reviews: 312,
+          rating: 0.30000001192092896,
+          imagea: 54,
+          imageb: 438,
+          imagec: 691,
+          imaged: 625,
+          tags: "party"
       }
-      ]
-
+      ],
+      withTags: [],
+      inSeries: []
     }
   }
 
+  setCurrentGameId() {
+    const idPath = window.location.pathname;
+    const urlID = idPath.match(/\d+/);
+    let newID = urlID[0];
+    newID = Number.parseInt(newID, 10);
+    this.setState({ gameID: newID });
+    console.log(newID);
+  }
+
   componentDidMount() {
-    axios.get('/api/getGameByID/' + this.props.gameid)
+    this.setCurrentGameId();
```

Comment:
Did you double check that changing the URL _actually_ loads new data when you change the page URL?

Based on my experience with React life cycles, `setCurrentGameId` won't be able to update `this.state.id` before the your first GET request grabs the id from state.

It may be working correctly since all your requests are within `componentDidMount()` but if you haven't already, just double check that the data is actually changing on ID change in the URL. If it isn't, you'll need to move your call on `this.setCurrentGameId()` from `componentDidMount()` to `componentDidUpdate`.